### DISCUSSION TITLE
Duckdb upsert is not working for all datatypes

### DISCUSSION THREAD

**Discussion by omidb** [duckdb/duckdb#8666] [General]
### What happens?

Right now upserts are not supported for some data types:
```
D CREATE TABLE t0(i INT PRIMARY KEY, friends VARCHAR[]);
D INSERT INTO t0 VALUES (1, ['person1']);
D INSERT OR REPLACE INTO t0 VALUES (1, ['person2']);
Error: Not implemented Error: List Update is not supported.
```
It would be great if duckdb can support it for every datatype

### To Reproduce

```
D CREATE TABLE t0(i INT PRIMARY KEY, friends VARCHAR[]);
D INSERT INTO t0 VALUES (1, ['person1']);
D INSERT OR REPLACE INTO t0 VALUES (1, ['person2']);
Error: Not implemented Error: List Update is not supported.
```

### OS:

Mac

### DuckDB Version:

v0.8.1 6536a77232

### DuckDB Client:

CLI D .version SQLite v0.8.1 6536a77232 clang-14.0.3

### Full Name:

Omid 

### Affiliation:

Personal

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by StLeoX** [comment]
Btw, why "['person12']" is recognized as a VARCHAR?
![2023-08-01_16-25-28](https://github.com/duckdb/duckdb/assets/59193798/78922237-0dbe-47cc-b233-3a23c379cdbe)

**Comment by omidb** [comment]
> Btw, why "['person12']" is recognized as a VARCHAR? ![2023-08-01_16-25-28](https://user-images.githubusercontent.com/59193798/257453080-78922237-0dbe-47cc-b233-3a23c379cdbe.png)

you have defined the friends column as VARCHAR. I defined it as VARCHAR[]

**Comment by StLeoX** [comment]
I mean this is another problem.

**Comment by szarnyasg** [comment]
At the moment, this is not yet supported.

**Reply by lpfy** [reply]
Hi, do you mean upsert not support for list data type OR list data type can not be updated? I am experiencing problem for common update query to update list type column.

`UPDATE t0 SET friends = ['p1','p2'] WHERE i = 1;`

return error:

`Constraint Error: Duplicate key "i: 1" violates primary key constraint. If this is an unexpected constraint violation please double check with the known index limitations section in our documentation (https://duckdb.org/docs/sql/indexes).`

I have read the documents, however, I am not updating the primary key column, the column with list data type is causing the problem.

**Comment by andriichumak** [comment]
Hi. Are there any plans to mitigate this?

**Comment by leonardovida** [comment]
Hi @szarnyasg! Following up on this, I've just tried with 1.1.4 (nightly) but the update List is still not supported. Any plans for this or official workarounds? Thank you!

**Comment by Mytherin** [comment]
This has been implemented in https://github.com/duckdb/duckdb/pull/15092

```sql
D CREATE TABLE t0(i INT PRIMARY KEY, friends VARCHAR[]);
D INSERT INTO t0 VALUES (1, ['person1']);
D INSERT OR REPLACE INTO t0 VALUES (1, ['person2']);
D FROM t0;
┌───────┬───────────┐
│   i   │  friends  │
│ int32 │ varchar[] │
├───────┼───────────┤
│   1   │ [person2] │
└───────┴───────────┘

```
